things/things.go

```diff
@@ -28,13 +28,10 @@ type ThingsPage struct {
 
 // ThingRepository specifies a thing persistence API.
 type ThingRepository interface {
-	// Save persists the thing. Successful operation is indicated by non-nil
+	// Save persists multiple things. Things are saved using a transaction. If one thing
+	// fails then none will be saved. Successful operation is indicated by non-nil
 	// error response.
-	Save(context.Context, Thing) (string, error)
-
-	// BulkSave persistest multiple things. Successful operation is indicated by non-nil
-	// error response.
-	BulkSave(context.Context, []Thing) ([]Thing, error)
+	Save(context.Context, []Thing) ([]Thing, error)
```

Comment:
Why are you not using variadic functions here too? (Same goes for channels)

Reply:
I was thinking at this point the things/channels would all be in a slice already and it wouldn't be needed.